sda/service.go

```diff
@@ -318,84 +303,69 @@ func (s *serviceManager) serviceByID(id ServiceID) (Service, bool) {
 	return serv, true
 }
 
-// ClientRequest is a generic packet to represent any kind of request a Service is
-// ready to process. It is simply a JSON packet containing two fields:
-// * Service: a string representing the name of the service for whom the packet
-// is intended for.
-// * Data: contains all the information of the request
-type ClientRequest struct {
-	// Name of the service to direct this request to
-	Service ServiceID
-	// Data containing all the information in the request
-	Data []byte
-}
-
-// ClientRequestID is the type that registered by the network library
-var ClientRequestID = network.RegisterPacketType(ClientRequest{})
-
-// CreateClientRequest creates a Request message out of any message that is
-// destined to a Service. XXX For the moment it uses protobuf, as it is already
-// handling abstract.Scalar/Public stuff that json can't do. Later we may want
-// to think on how to change that.
-func CreateClientRequest(service string, r interface{}) (*ClientRequest, error) {
-	sid := ServiceFactory.ServiceID(service)
-	log.Lvl1("Name", service, " <-> ServiceID", sid.String())
-	buff, err := network.MarshalRegisteredType(r)
-	if err != nil {
-		return nil, err
-	}
-	return &ClientRequest{
-		Service: sid,
-		Data:    buff,
-	}, nil
-}
-
 // Client is a struct used to communicate with a remote Service running on a
 // sda.Conode
 type Client struct {
```

Comment:
I didn't want to have a map of open connections, so I keep only one open at any time. If the client connects to different conodes, the old connection will be closed and a new opened.